internal/kafka/internal/config/dynamic_scaling_config_test.go

```diff
@@ -3,324 +3,503 @@ package config
+				KafkaWorkloadPerInstanceType: nil,
+			},
+			args: args{
+				cloudProvider: cloudproviders.AWS,
+			},
+			wantErr: true,
+		},
+		{
+			name: "should return an error when ClusterWideWorkload is invalid when min nodes is less than 3",
+			fields: fields{
+				ClusterWideWorkload: &ComputeMachineConfig{
+					ComputeMachineType: "some-machine",
+					ComputeNodesAutoscaling: &ComputeNodesAutoscalingConfig{
+						MinComputeNodes: 1, // This will cause a validation error as min should be 3
+						MaxComputeNodes: 9,
+					},
+				},
+				KafkaWorkloadPerInstanceType: map[string]ComputeMachineConfig{
+					"type": {
+						ComputeMachineType: "machine-type",
+						ComputeNodesAutoscaling: &ComputeNodesAutoscalingConfig{
+							MaxComputeNodes: 1,
+							MinComputeNodes: 1,
+						},
+					},
+				},
+			},
+			args: args{
+				cloudProvider: cloudproviders.AWS,
+			},
+			wantErr: true,
+		},
+		{
+			name: "should return an error when ClusterWideWorkload is invalid when marshalled configurations are missing",
+			fields: fields{
+				ClusterWideWorkload: &ComputeMachineConfig{
+					ComputeMachineType: "",
+					ComputeNodesAutoscaling: &ComputeNodesAutoscalingConfig{
+						MinComputeNodes: 0,
+						MaxComputeNodes: 0,
+					},
+				},
+				KafkaWorkloadPerInstanceType: map[string]ComputeMachineConfig{
+					"type": {
+						ComputeMachineType: "machine-type",
+						ComputeNodesAutoscaling: &ComputeNodesAutoscalingConfig{
+							MaxComputeNodes: 1,
+							MinComputeNodes: 1,
 						},
 					},
-					"instance-type-2": {
-						ComputeNodesConfig: &DynamicScalingComputeNodesConfig{
+				},
+			},
+			args: args{
+				cloudProvider: cloudproviders.AWS,
+			},
+			wantErr: true,
+		},
+		{
+			name: "should return an error when KafkaWorkloadPerInstanceType is invalid",
+			fields: fields{
+				ClusterWideWorkload: &ComputeMachineConfig{
+					ComputeMachineType: "some-machine",
+					ComputeNodesAutoscaling: &ComputeNodesAutoscalingConfig{
+						MinComputeNodes: 3, // This will cause a validation error as min should be 3
+						MaxComputeNodes: 9,
+					},
+				},
+				KafkaWorkloadPerInstanceType: map[string]ComputeMachineConfig{
+					"type": {
+						ComputeMachineType: "machine-type",
+						ComputeNodesAutoscaling: &ComputeNodesAutoscalingConfig{
 							MaxComputeNodes: 1,
+							MinComputeNodes: 0,
 						},
 					},
 				},
 			},
-			want: map[string]InstanceTypeDynamicScalingConfig{
-				"instance-type": {
-					ComputeNodesConfig: &DynamicScalingComputeNodesConfig{
-						MaxComputeNodes: 10,
+			args: args{
+				cloudProvider: cloudproviders.AWS,
+			},
+			wantErr: true,
+		},
+		{
+			name: "should not return an error when valid configurations are provided",
+			fields: fields{
+				ClusterWideWorkload: &ComputeMachineConfig{
+					ComputeMachineType: "some-machine",
+					ComputeNodesAutoscaling: &ComputeNodesAutoscalingConfig{
+						MinComputeNodes: 3, // This will cause a validation error as min should be 3
```

Comment:
Same as https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pull/1330/files#r985615924

Reply:
This comment is misplaced due to a copy/paste issue of the config. Will remove.